### PR TITLE
feat: add candidate discard and reset modal. Application of activity …

### PR DESCRIPTION
**Pull request by hatdogguldo**
…logs on required module pages.

### CODE REVIEW

File: src/add-candidate.php [hunk 1]
```diff
@@ -42,7 +42,73 @@
...
+                        font-size: 1.4rem;
+                        color: #aaa;
+                        opacity: 1;
+                    }
+
+                    .close:hover {
+                        color: #000;
+                        opacity: 1;
+                    }
+                </style>
```
Comment (Carl-Tabuso) [inline src/add-candidate.php]: Can you have this css in a separate or existing stylesheet file?

File: src/add-candidate.php [hunk 2]
```diff
@@ -514,6 +601,118 @@ function duplicateForm() {
...
+                    });
+
+                    // Handle form reset confirmation
+                    document.getElementById("confirmReset").addEventListener("click", function() {
+                        isFormDirty = false;
+                        form.reset();
+                        resetModal.hide();
+                    });
+                });
+            </script>
```
Comment (Carl-Tabuso) [inline src/add-candidate.php]: Put this in a separate or existing .js file

**Review by Carl-Tabuso — CHANGES_REQUESTED** [review]
I'm encountering a page reload upon adding a new candidate. This is due to a setTimeOut function you have in your javascript somewhere. It would be best to replace it with a toast so it won't be sort of disruptive that they have to click or close the modal and carry on with adding additional candidates and such.

https://github.com/user-attachments/assets/4f39af70-247f-4c5b-b75d-688a2e8f633d

**Comment by Carl-Tabuso** [comment]
You may also address here the [issue ticket](https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/issues/254) regarding the security issue detected by code scanning alert.

File: src/add-candidate.php (hunk 2 above)
Comment (hatdogguldo) [reply src/add-candidate.php]: fixed this with a new file name: candidate-modal.js

File: src/add-candidate.php (hunk 1 above)
Comment (hatdogguldo) [reply src/add-candidate.php]: I tried to separate it and put it in the existing stylesheet. But it wouldn't function if it was separated. Could you suggest any techniques, @Carl-Tabuso ?

**Comment by hatdogguldo** [comment]
I comment out the setTimeout so that when the user clicks on the background, it will just remove the modal. I retain it as modal because that is on the UI. 
image

**Review by Carl-Tabuso — APPROVED** [review]
Have your QA thorouhgly test this.

**Pull request merged**